includes/class-magic-link.php

```diff
@@ -729,39 +754,40 @@ function () {
 				1
 			);
 		}
-
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		if ( ! isset( $_GET['action'] ) || self::AUTH_ACTION !== $_GET['action'] ) {
 			return;
 		}
-
 		$errored = false;
+		$user    = false;
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		if ( ! isset( $_GET['token'] ) || ! isset( $_GET['email'] ) ) {
+		if ( ! isset( $_GET['token'] ) || ! isset( $_GET['secret'] ) ) {
 			$errored = true;
 		}
-
 		if ( ! $errored ) {
 			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-			$email = \sanitize_email( $_GET['email'] );
-			if ( $email ) {
-				$user = \get_user_by( 'email', $email );
-				if ( ! $user ) {
+			$secret = filter_input( INPUT_GET, 'secret', FILTER_SANITIZE_STRING );
+			if ( $secret ) {
+				$users = \get_users(
+					[
+						'meta_key'   => self::USER_SECRET_META,
+						'meta_value' => $secret, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+					]
+				);
+				if ( empty( $users ) || count( $users ) > 1 ) {
```

Comment:
Non-blocking performance tweak:

```suggestion
				$user_query = new \WP_User_Query(
					[
						'meta_key'    => self::USER_SECRET_META,
						'meta_value'  => $secret, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
						'number'      => 1,
						'count_total' => false,
					]
				);
				$users = $user_query->get_results();
				if ( empty( $users ) ) {
```

Reply:
Thanks for the suggestion! Updated in https://github.com/Automattic/newspack-plugin/pull/3476/commits/81af75c89b982391bd23766d67ede80822d4ea10